### PR TITLE
cleanup: remove redundant network port identifiers from maker logs an…

### LINKED ISSUE #798: feat: Remove network as maker identifier
## Problem Statement

Currently the network port of the maker doesn't serve any functional purpose. Its only used as identifier in the logs. That's redundant.

## Proposed Solution

Find a better identifier and remove network ports from all places.

Same as taker, it doesn't need network port for anything. 

## Affected Component(s)
- [x] **makerd** (background daemon)
- [x] **taker** (CLI client)
- [x] **maker-cli** (command-line tool)
- [x] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

## Protocol Version
- [ ] Legacy (ECDSA)
- [ ] Taproot-Musig2
- [x] Both
- [ ] New protocol addition
- [ ] Not applicable

### PR DESCRIPTION
**Pull request by Abhishek2634**
# Pull Request

## Description
Fixed redundant logging identifiers where the network port was used as a prefix. Removed the port field from ThreadPool as it was only used for logging.

Closes #798 

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Fix Screenshot
**Integration tests passed**
Screenshot 2026-03-14 at 3 42 35 PM

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [ ] Both
- [ ] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [x] All unit tests pass (`cargo test`)
- [x] Integration tests pass (`cargo test --features integration-test`)
- [x] Changes were manually tested on **regtest**
- [x] End-to-end maker ↔ taker swap tested (where applicable)
- [x] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## How to Test

```bash
# Standard swap (good baseline)
cargo test --test standard_swap --features integration-test -- --nocapture

# Abort scenarios
cargo test --test abort1 --features integration-test -- --nocapture
cargo test --test taproot_taker_abort1 --features integration-test -- --nocapture

# Malicious behavior & recovery
cargo test --test malice1 --features integration-test -- --nocapture
cargo test --test taproot_timelock_recovery --features integration-test -- --nocapture

# Or run the full suite
cargo test --features integration-test
```

## Summary by CodeRabbit

* **Refactor**
  * Standardized and simplified logs across the maker module: removed per-port prefixes and now show peer, swap, or contract contexts with clearer thread/join messages.

* **New Features**
  * Added private-key handover handling in the maker message flow.
  * Added a sweep action to sweep incoming swapcoins after successful coinswaps.

* **Chores**
  * Startup now logs the Bitcoin network; connection logs include remote peer addresses; minor formatting and verbosity cleanups.

### CODE REVIEW

**Comment by Abhishek2634** [comment]
Hi  @mojoX911

I have completed the refactoring for the issue regarding redundant network port identifiers. Here was my approach to solving it:

Remove the redundant port field from ThreadPool in api.rs and api2.rs.
Strip out [{}] and network_port arguments from all log macros across the Maker crates.
remove all "argument never used" errors and verified everything with the standard_swap integration test.

I have done runnig `cargo check `and `cargo test --features=integration-test --test standard_swap -- --nocapture` . and all test passed here. I have attached the screenshot of test results above in this PR.

Thanks :)

File: src/maker/api.rs [hunk 1]
```diff
@@ -619,18 +606,14 @@ pub(crate) fn check_for_broadcasted_contracts(maker: Arc<Maker>) -> Result<(), M
...
                         );
                         failed_swap_ip.push(ip.clone());
 
                         // Spawn a separate thread to wait for contract maturity and broadcasting timelocked/hashlocked.
                         let maker_clone = maker.clone();
-                        log::info!(
-                            "[{}] Spawning recovery thread after seeing contracts in mempool",
-                            maker.config.network_port
-                        );
+                        log::info!("Spawning recovery thread after seeing contracts in mempool",);
```
[the remark this answers is not in the document]
Comment (Abhishek2634) [reply src/maker/api.rs:621]: done

File: src/maker/api.rs [hunk 2]
```diff
@@ -884,9 +879,14 @@ fn recover_via_timelock(
...
+            .map(|c| c.contract_tx.compute_txid().to_string())
+            .unwrap_or_default();
         log::info!(
-            "[{}] Maker timelock recovery: {}/{} txs broadcasted",
-            maker.config.network_port,
+            "[Contract: {}] Maker timelock recovery: {}/{} txs broadcasted",
+            anchor_id,
             broadcasted.len(),
             outgoing.len()
         );
```
[the remark this answers is not in the document]
Comment (Abhishek2634) [reply src/maker/api.rs:894]: done

**Comment by Abhishek2634** [comment]
@coderabbitai review
